ddl/db_partition_test.go

```diff
@@ -825,7 +825,7 @@ func (s *testIntegrationSuite5) TestAlterTableAddPartitionByListColumns(c *C) {
 			ddl.ErrMultipleDefConstInListPart,
 		},
 		{"alter table t add partition (partition p6 values in (('a','a')))",
-			ddl.ErrNotAllowedTypeInPartition,
+			ddl.ErrWrongTypeColumnValue,
```

Comment:
@bb7133 @blacktear23  sorry, this test case show that we have do this check in `alter` statement. 😥

Reply:
I have test this in mysql, this fix result is same as mysql.

Reply:
below is mysql result:

```sql
mysql> create table t (id int, name varchar(10)) partition by list columns (id,name) (
    ->     partition p0 values in ((1,'a'),(2,'b')),
    ->     partition p1 values in ((3,'a'),(4,'b')),
    ->     partition p3 values in ((5,null))
    -> );
Query OK, 0 rows affected (0.03 sec)

mysql> alter table t add partition (partition p6 values in (('a','a')));
ERROR 1654 (HY000): Partition column values of incorrect type
```

Reply:
By the way seems there may have double check for type on `alter table add partition` logic. Shall we need to find it out and remove the logic? @bb7133

Reply:
> By the way seems there may have double check for partition values type 

there are many double-check, even triple check  😂, a lot of work should be done.